domain-models-runtime/src/main/java/org/folio/rest/impl/JsonSchemaAPI.java

```diff
@@ -0,0 +1,168 @@
+            GetJsonSchemaResponse.respond500WithTextPlain(e.getMessage())
+          )
+        );
+      }
+    });
+  }
+
+  @Validate
+  @Override
+  public void getJsonSchemaByName(
+    String name,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext
+  ) {
+    String okapiUrl = okapiHeaders.get("x-okapi-url");
+    vertxContext.runOnContext(v -> {
+      try {
+        JsonNode schemaNode = getSchemaByName(name, okapiUrl);
+        if(schemaNode != null) {
+          asyncResultHandler.handle(
+            Future.succeededFuture(
+              GetJsonSchemaByNameResponse.respond200WithApplicationJson(schemaNode)
+            )
+          );
+        } else {
+          String notFoundMessage = "Schema " + name + " not found";
+          asyncResultHandler.handle(
+            Future.succeededFuture(
+              GetJsonSchemaByNameResponse.respond404WithTextPlain(notFoundMessage)
+            )
+          );
+        }
+      } catch (Exception e) {
+        log.error(e.getMessage(), e);
+        asyncResultHandler.handle(
+          Future.succeededFuture(
+            GetJsonSchemaByNameResponse.respond500WithTextPlain(e.getMessage())
+          )
+        );
+      }
+    });
+  }
+
+  private List<String> getSchemas() throws IOException {
+    List<String> schemas = new ArrayList<>();
+    File jarFile = new File(getClass().getProtectionDomain().getCodeSource().getLocation().getPath());
+    JarFile jar = new JarFile(jarFile);
+    Enumeration<JarEntry> entries = jar.entries();
+    while(entries.hasMoreElements()) {
+      JarEntry entry = entries.nextElement();
+      String entryName = entry.getName();
+      if (entryName.startsWith("ramls/") && entryName.endsWith(".json")) {
+        String schemaPath = entryName.substring(6);
+        if(!schemaPath.contains("/")) {
+          String schemaName = schemaPath.substring(schemaPath.lastIndexOf("/") + 1);
+          try {
+            InputStream is = jar.getInputStream(entry);
+            ObjectMapperTool.getMapper().readValue(is, JsonNode.class);
+            is.close();
+            schemas.add(schemaName);
+          } catch(Exception e) {
+            log.info("{} is not a valid json file", entryName);
+          }
+        }
+      }
+    }
+    jar.close();
+    return schemas;
+  }
+
+  private JsonNode getSchemaByName(String name, String okapiUrl) throws IOException {
+    JsonNode schemaNode = null;
+    File jarFile = new File(getClass().getProtectionDomain().getCodeSource().getLocation().getPath());
+    JarFile jar = new JarFile(jarFile);
+    Enumeration<JarEntry> entries = jar.entries();
+    while(entries.hasMoreElements()) {
+      JarEntry entry = entries.nextElement();
+      String entryName = entry.getName();
+      if (entryName.startsWith("ramls/")) {
+        String schemaName = entryName.substring(entryName.lastIndexOf("/") + 1);
+        if (schemaName.equals(name)) {
```

Comment:
How do you distinguish ramls/instance.json and ramls/raml-util/schemas/codex/instance.json in https://github.com/folio-org/mod-inventory-storage/tree/master/ramls ?

Reply:
I assumed uniqueness of JSON Schema file names. I will address this.

Reply:
http://localhost:9130/jsonSchemas?path=instance.json

http://localhost:9130/jsonSchemas?path=raml-util/schemas/codex/instance.json